Backend/controllers/StatisticsResourceController.js

```diff
@@ -32,46 +31,41 @@ exports.getStatisticsResourceById = async(req, res) => {
 };
 
 exports.getStatisticsResourceByFields = async(req, res) => {
-    try {
-        if (req.query.language != null && req.query.title != null) {
-            let statisticsResource = await StatisticsResource.find({ language: req.query.language, title: req.query.title });
-            res.status(200).send(statisticsResource);
-        } else if (req.query.language != null) {
-            let statisticsResource = await StatisticsResource.find({ language: req.query.language });
-            res.status(200).send(statisticsResource);
-        } else if (req.query.title != null) {
-            let statisticsResource = await StatisticsResource.find({ title: req.query.title,language:"English" });
-            res.status(200).send(statisticsResource);
-        } else {
-            let statisticsResource = await StatisticsResource.find({language:"English"});
-            res.status(200).send(statisticsResource);
-        }
 
-    } catch (error) {
-        res.status(500).send(error);
+    if (req.query.language != null && req.query.title != null) {
+        let statisticsResource = await StatisticsResource.find({ language: req.query.language, title: req.query.title });
+        res.status(200).send(statisticsResource);
+    } else if (req.query.language != null) {
+        let statisticsResource = await StatisticsResource.find({ language: req.query.language });
+        res.status(200).send(statisticsResource);
+    } else if (req.query.title != null) {
+        let statisticsResource = await StatisticsResource.find({ title: req.query.title,language:"English" });
+        res.status(200).send(statisticsResource);
+    } else {
+        let statisticsResource = await StatisticsResource.find({language:"English"});
+        res.status(200).send(statisticsResource);
     }
+
 };
 
-// - [DEPRECATED: This endpoint is not in use (not necessary)]
+
 exports.updateStatisticsResource = async(req, res) => {
     try {
-        let statisticsResource = await StatisticsResource.findOneAndUpdate(req.params._id, req.body);
-        if (statisticsResource) {
-            let updatedStatisticsResource = await StatisticsResource.findById(statisticsResource._id);
-            res.send(updatedStatisticsResource);
-        } else {
-            res.send("stat resource not found");
-        }
+        await StatisticsResource.findOneAndUpdate(req.params.id, req.body);
```

Comment:
You might be looking for findByIdAndUpdate since findOneAndUpdate accepts object not string